src/mimir_coordinator.py

```diff
@@ -120,10 +120,20 @@ def build_config(self, _charm_config: Dict[str, Any]) -> Dict[str, Any]:
         }
 
         # todo: TLS config for memberlist
-        if self._tls_requirer:
-            mimir_config["tls_enabled"] = True
-            mimir_config["tls_cert_path"] = self._tls_requirer.cacert
-            mimir_config["tls_key_path"] = self._tls_requirer.key
-            mimir_config["tls_ca_path"] = self._tls_requirer.capath
+        if tls:
+            mimir_config["server"] = {
+                "http_tls_config": {
+                    "cert": self._tls_requirer.cert,
+                    "key": self._tls_requirer.key,
```

Comment:
- Does this mean the private key will travel over relation data?
- With nginx in place, are we going for e2e tls or just termination?

Reply:
The idea is that we create the config in the coordinator and distribute it to the workers. That's why we are setting everything here.

Now that you mention this, I think that if the communication channel where relation data travels is encrypted, then there is no risk. WDYT?

If we cannot ensure this the process of generating the configuration will be a bit more complex.

As far as I understand for nginx we need e2e, anyway @PietroPasotti or @lucabello may have better insights

Reply:
> if the communication channel where relation data travels is encrypted

That is one of the reasons we're moving away from peer data for storing secrets.